tf-modules/archive/sns2elasticsearch.tf

```diff
@@ -19,18 +20,36 @@ resource "aws_lambda_function" "sns2elasticsearch" {
   environment {
     variables = {
       CMR_ENVIRONMENT = var.cmr_environment
-      GranulesTable   = var.dynamo_tables.Granules
-      ExecutionsTable = var.dynamo_tables.Executions
-      PdrsTable       = var.dynamo_tables.Pdrs
+      GranulesTable   = var.dynamo_tables.granules.name
+      ExecutionsTable = var.dynamo_tables.executions.name
+      PdrsTable       = var.dynamo_tables.pdrs.name
       ES_HOST         = var.elasticsearch_hostname
       stackName       = var.prefix
     }
   }
-  tags = {
-    Project = var.prefix
-  }
+  tags = merge(local.default_tags, { Project = var.prefix })
   vpc_config {
     subnet_ids         = var.lambda_subnet_ids
     security_group_ids = [aws_security_group.no_ingress_all_egress.id]
   }
 }
+
+# Step Function Tracker
+
+resource "aws_sns_topic" "sftracker" {
+  name = "${var.prefix}-sftracker"
+  tags = local.default_tags
+}
+
+resource "aws_sns_topic_subscription" "sftracker_to_sns2elasticsearch" {
+  endpoint  = aws_lambda_function.sns2elasticsearch.arn
+  protocol  = "lambda"
+  topic_arn = aws_sns_topic.sftracker.arn
+}
+
+resource "aws_lambda_permission" "sftracker_to_sns2elasticsearch" {
+  action        = "lambda:InvokeFunction"
+  function_name = aws_lambda_function.sns2elasticsearch.arn
+  principal     = "sns.amazonaws.com"
+  source_arn    = aws_sns_topic.sftracker.arn
+}
```

Comment:
I assume these will eventually be replaced by the workflow reporting tf module, @markdboyd?

Reply:
Yes, this specific lambda is irrelevant once the TF deployment is setup to use the new SNS consumers. I'm not sure what ticket that would be under though.

I actually already have the whole workflow reporting module ready to go as part of 1394, so I could drop it as part of my work now? I couldn't really test it until we get Marc's work on the example module merged though. Or I could leave integrating my workflow reporting module with the rest of the TF deployment for 1396

Reply:
Resolving, discussed in alternate channels.